django/contrib/staticfiles/storage.py

```diff
@@ -273,8 +285,57 @@ def path_level(name):
                         hashed_name = force_text(self.clean_name(saved_name))
 
                 # and then set the cache accordingly
-                hashed_files[self.hash_key(name)] = hashed_name
+                hashed_files[hash_key] = hashed_name
+
+                yield name, hashed_name, processed, substitutions
+
+    def post_process(self, paths, dry_run=False, **options):
```

Comment:
It might be nice to reorder this above _post_process so that lots of code doesn't appear as changed when it was only moved. It would certainly ease review.